compiler/src/iree/compiler/Codegen/Dialect/Codegen/IR/IREECodegenInterfaces.td

```diff
@@ -112,6 +112,19 @@ def IREECodegen_LoweringConfigAttrInterface :
       /*defaultImplementation=*/[{
         return ::llvm::SmallVector<OpFoldResult>();
       }]
+    >,
+    InterfaceMethod<
+      /*desc=*/[{
+        Gets the name of the custom lowering strategy to apply to the annotated
+        operation.
+      }],
+      /*retTy=*/"::std::optional<::llvm::StringRef>",
```

Comment:
Do we want this to be a `StringRef` or an actualy `SymbolRef`.
I guess making a `SymbolRef` will make it impossible to the "external TD script" thing. But I find it a bit spooky/action-at-a-distance. Someone has to set a lowering config that magically knows the name of the transform sequence that happens to be in the external module. I dont know if we really want to support that case. Unless you have a strong reason for it, just using a `NestedSymbolRef` might give more gaurantees and make it an easier lookup?

Reply:
Actually `NestedSymbolRef` is ok here, I can change that. This is just the getter, and the symbol must be resolvable by the time we get here. The actual symbol can be stored as a string if needed still.

Re: external modules. We need this because codegen pipelines are nested on functions, meaning we can no longer manipulate symbols with the same nesting by the time we get here. This matters because for the most basic case I want to support (hand authored external functions) we have to inline because we can't bufferize or distribute across function boundaries today. The other option is to import any callable functions in a separate nested module (which I tried) but currently `util.call` only supports flat symbol refs so it's a bit of a pain to support cleanly.